cmd/temporalite/main_integ_test.go

```diff
@@ -0,0 +1,138 @@
+// MIT License
+//
+// Copyright (c) 2022 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2021 Datadog, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package main_test
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"syscall"
+	"testing"
+	"time"
+
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/client"
+)
+
+func TestMTLSConfig(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// Run ephemerally in the mtls example folder using the config there. This
+	// does expect port 7233 to be free.
+	_, thisFile, _, _ := runtime.Caller(0)
+	args := []string{
+		"run", filepath.Dir(thisFile),
+		"start",
+		"--ephemeral",
+		"--config", ".",
+		"--namespace", "default",
+		"--log-format", "pretty",
+	}
+	if !testing.Verbose() {
+		args = append(args, "--log-level", "warn")
+	}
+	cmd := exec.CommandContext(ctx, "go", args...)
+	mtlsDir := filepath.Join(thisFile, "../../../internal/examples/mtls")
+	cmd.Dir = mtlsDir
+	cmd.Stdout, cmd.Stderr = os.Stdout, os.Stderr
+	t.Logf("Running go with args %v in %v", args, cmd.Dir)
+	if err := cmd.Start(); err != nil {
+		log.Fatal(err)
```

Comment:
Is this necessary vs. using `t.Fatal`? I'm not sure if this would interfere with tests running in other packages.

Reply:
Sorry, thought I was using that, mistyped everywhere, will fix

Reply:
FIxed